lib/components/PoolUI.jsx

```diff
@@ -41,44 +62,49 @@ export const PoolUI = (props) => {
   const provider = walletContext.state.provider
   const usersAddress = walletContext._onboard.getState().address
 
-  const [ethBalance, setEthBalance] = useState(ethers.utils.bigNumberify(0))
-  const [poolAddresses, setPoolAddresses] = useState({
-    prizePool,
-  })
-  const [genericChainValues, setGenericChainValues] = useState({
-    loading: true,
-    tokenSymbol: 'TOKEN',
-    poolTotalSupply: '1234',
-  })
-
-  const [usersChainValues, setUsersChainValues] = useState({
-    loading: true,
-    usersTicketBalance: ethers.utils.bigNumberify(0),
-    usersTokenAllowance: ethers.utils.bigNumberify(0),
-    usersTokenBalance: ethers.utils.bigNumberify(0),
-  })
+  const [ethBalance, setEthBalance] = useAtom(ethBalanceAtom)
+  const [prizePoolType] = useAtom(prizePoolTypeAtom)
+  const [poolAddresses, setPoolAddresses] = useAtom(poolAddressesAtom)
+  const [_usersAddress, setUsersAddress] = useAtom(usersAddressAtom)
+  const [erc20Awards, setErc20Awards] = useAtom(erc20AwardsAtom)
+  const [network, setNetwork] = useAtom(networkAtom)
+  const [poolChainValues, setPoolChainValues] = useAtom(poolChainValuesAtom)
+  const [userChainValues, setUserChainValues] = useAtom(userChainValuesAtom)
 
-  useInterval(() => {
-    fetchChainData(
-      networkName,
-      usersAddress,
-      poolAddresses,
-      setPoolAddresses,
-      setGenericChainValues,
-      setUsersChainValues
-    )
-  }, 25000)
+  useEffect(() => {
+    // TODO: Probably need to reset other atoms if this changes.
+    setPoolAddresses({
+      prizePool
+    })
+  }, [prizePool])
+
+  useEffect(() => {
+    setUsersAddress(usersAddress)
+  }, [usersAddress])
 
   useEffect(() => {
-    fetchChainData(
-      networkName,
-      usersAddress,
+    setNetwork({
+      name: networkName,
+      id: nameToChainId(networkName)
+    })
+  }, [networkName])
+
+  usePrizePoolType()
```

Comment:
hmm, I'm not sure how I feel about these implicit hooks. The only pattern I've seen with hooks are ones that return something, and then you consume the returned values in that component. But it is good to know that we can use hooks inside other hooks, that's awesome. For some reason I thought I tried that the other day and react complained about it.

Reply:
Yeah I guess it would make sense to return the values that are being updated from jotai here, that would fit the pattern a little better.
I've definitely used hooks before that don't necessarily return any values, ex. https://usehooks.com/useOnClickOutside/.
Maybe you had a hook behind a conditional or something, there's plenty of rules but combining them is encouraged.

Reply:
Okay cool, let's keep it as it is.